tests/editing/totals_and_components/test_totals_and_components.py

```diff
@@ -6378,17 +6527,64 @@ def test_correct_components(
             )
 
 
-def compare_list_of_floats(components: List[float], expected_components: List[float]):
+def compare_results_to_expected_results(
+    results: TotalsAndComponentsOutput, expected_result: tuple
+):
+    assert results.identifier == expected_result[0]
+
+    if results.absolute_difference is None:
+        assert results.absolute_difference == expected_result[1]
+    else:
+        assert round(results.absolute_difference, 7) == round(expected_result[1], 7)
+
+    if results.low_percent_threshold is None:
+        assert results.low_percent_threshold == expected_result[2]
+    else:
+        assert round(results.low_percent_threshold, 7) == round(expected_result[2], 7)
+
+    if results.high_percent_threshold is None:
+        assert results.high_percent_threshold == expected_result[3]
+    else:
+        assert round(results.high_percent_threshold, 7) == round(expected_result[3], 7)
+
+    if results.final_total is None:
+        assert results.final_total == expected_result[4]
+    else:
+        assert round(results.final_total, 7) == round(expected_result[4], 7)
+
+    assert results.tcc_marker == expected_result[6]
+
+    if results.tcc_marker == "T" or results.tcc_marker == "C":
+        sum_of_components = 0
+        for component in results.final_components:
+            print(f"fc - {results.final_components}")
+            if not math.isnan(component):
+                sum_of_components += Decimal(str(component))
+
+        assert round(sum_of_components, 7) == round(expected_result[4], 7)
+
+    compare_list_of_decimals(results.final_components, expected_result[5])
+
+
+def compare_list_of_decimals(
+    components: List[Decimal], expected_components: List[Decimal]
+):
     for i, (component, expected) in enumerate(zip(components, expected_components)):
         if math.isnan(component) and math.isnan(expected):
-            assert math.isnan(component) and math.isnan(expected), f"Both components at index {i} are NaN"
+            assert math.isnan(component) and math.isnan(
+                expected
+            ), f"Both components at index {i} are NaN"
         elif math.isnan(component) or math.isnan(expected):
-            assert False, f"Component at index {i} is NaN, the other is not component {component}, expected {expected}"
-        elif component != expected:
-            assert False, f"Values of components at index {i} do not match: component {component}, expected {expected}"
+            assert (
+                False
+            ), f"Component at index {i} is NaN, the other is not component {component}, expected {expected}"
+        elif round(component, 7) != round(expected, 7):
+            assert (
+                False
+            ), f"Values of components at index {i} do not match: component {component}, expected {expected}"
 
 
-def calculate_weighted(total: float, components: List[float]):
+def calculate_weighted(total: Decimal, components: List[Decimal]):
```

Comment:
Do we use this function anywhere? When we were converting from Decimal to float it made sense but now I am not sure we need it.